src/imap/mod.rs

```diff
@@ -574,6 +574,9 @@ impl Imap {
                     match session.fetch(set, JUST_UID).await {
                         Ok(mut list) => {
                             if let Some(Ok(msg)) = list.next().await {
+                                while let Some(_response) = list.next().await {
```

Comment:
if you use `async-std@master` should be able to do the following instead

```rust
let uids = list.collect::<Vec<_>>()?;
```

I believe, which will be nicer and make it easier to ensure all responses are read.

Reply:
`collect` looks like it will allocate memory just to throw responses away. And it does not work with our current version of `async-std`, so I will leave current version as is, especially since I have already tested it.

Reply:
Collect also did not work for me due to some `Send` issues. And now the code processes all the responses anyway, filtering through unsolicited responses.